### PR TITLE
Handle gates with ctrl_stete=0

### LINKED ISSUE #2127: qc.cx(0, 1, ctrl_state=0) fails - AerError: 'unknown instruction: cx_o0'
### Informations

- **Qiskit Aer version**: 0.14.1
- **Python version**: 3.11
- **Operating system**: Mac

### What is the current behavior?
New version of AER does not recognize cx gate with control state = 0, while it did recognize cnot with control state = 0, which was not removed in favor of cx in 0.1. 

### Steps to reproduce the problem

```
qc = QuantumCircuit(2)
qc.cx(0, 1) # works
qc.cx(0, 1, ctrl_state=1) # works
qc.cx(0, 1, ctrl_state=0) # fails - AerError: 'unknown instruction: cx_o0'
qc.measure_all()

simulator = AerSimulator()
simulator.run(qc).result().get_counts()
```

### What is the expected behavior?
Should handle the `qc.cx(0, 1, ctrl_state=0) ` natively

### Suggested solutions
Should be easy since we already had in there cnot with control on 0, just rename mapping

### PR DESCRIPTION
**Pull request by doichanj**
### Summary

This PR fixes #2127

### Details and comments

This PR compiles gates including ctrl_state=0 before passing C++ simulator
For example, `cx(0, 1, ctrl_state=0) `is compiled into `x(0)`, `cx(0,1)` and `x(0)` gates

### CODE REVIEW

File: qiskit_aer/backends/aer_compiler.py [hunk 1]
```diff
@@ -823,6 +823,14 @@ def _assemble_op(
...
         aer_circ.gate(name, qubits, params, [], conditional_reg, aer_cond_expr,
                       label if label else name)
+    elif name == "cx_o0":
+        qubits_0 = [qubits[0]]
+        aer_circ.gate("x", qubits_0, params, [], conditional_reg, aer_cond_expr,
+                      label if label else "x")
+        aer_circ.gate("cx", qubits, params, [], conditional_reg, aer_cond_expr,
+                      label if label else "cx")
+        aer_circ.gate("x", qubits_0, params, [], conditional_reg, aer_cond_expr,
+                      label if label else "x")
```
Comment (hhorii) [inline qiskit_aer/backends/aer_compiler.py:833]: need to update `num_of_aer_ops` to `3`.

File: qiskit_aer/backends/aer_compiler.py [hunk 2]
```diff
@@ -823,6 +823,14 @@ def _assemble_op(
     }:
         aer_circ.gate(name, qubits, params, [], conditional_reg, aer_cond_expr,
                       label if label else name)
+    elif name == "cx_o0":
```
Comment (atharva-satpute) [inline qiskit_aer/backends/aer_compiler.py:826]: Shouldn't we check this for other gates as well? The same would be applicable to `cy`, `cz` and other gates (with ctrl_state arg) right?

File: qiskit_aer/backends/aer_compiler.py [hunk 3]
```diff
@@ -823,6 +823,16 @@ def _assemble_op(
     }:
         aer_circ.gate(name, qubits, params, [], conditional_reg, aer_cond_expr,
                       label if label else name)
+    elif "_o0" in name:
+        orig_name = name.replace("_o0", "")
+        qubits_0 = [qubits[0]]
+        aer_circ.gate("x", qubits_0, params, [], conditional_reg, aer_cond_expr,
+                      label if label else "x")
+        aer_circ.gate(orig_name, qubits, params, [], conditional_reg, aer_cond_expr,
```
Comment (hhorii) [inline qiskit_aer/backends/aer_compiler.py:831]: I think we need to check `orig_name` is in supported gates in `AerCircuit`. In addition, `mc*` will use more variation of `ctrl_state` in addition to `_o0` (such as `_o0101`).  

I think `transpiler` is good to decompose gates with basis gates. Or we say "gate with `ctrl_state` is not supported".

**Comment by hhorii** [comment]
```
from qiskit import QuantumCircuit, transpile
from qiskit_aer import AerSimulator

qc = QuantumCircuit(3)
qc.x(0) # 0q is 1
qc.x(1) # 1q is 1
qc.mcx([0, 1], 2, ctrl_state="01") # 2q must be 0
qc.measure_all()

simulator = AerSimulator()

print("actual: ", simulator.run(qc).result().get_counts())
print("ideal: ", simulator.run(transpile(qc, simulator)).result().get_counts())
```

6918b90 shows inconsistent results.

**Review by hhorii — APPROVED** [review]
I confirmed that the above sample works well.

**Pull request merged**